components/builder-api/src/http/handlers.rs

```diff
@@ -176,6 +196,11 @@ pub fn accept_invitation(req: &mut Request) -> IronResult<Response> {
     }
 
     let mut conn = Broker::connect().unwrap();
+
+    if !try!(validate_origin_invitation(&mut conn, &request)) {
```

Comment:
Yes! All authentication and permissions should be checked on the gateway nodes and not piped through to the services themselves. We can *trust all* messages once they get past a gateway